code/datums/helper_datums/stack_end_detector.dm

```diff
@@ -0,0 +1,32 @@
+/**
+	Stack End Detector.
+	Can detect if a given code stack has exited, used by the mc for stack overflow detection.
+
+ **/
+/datum/stack_end_detector
+	var/datum/weakref/_WF
+	var/datum/stack_canary/_canary
+
+/datum/stack_end_detector/New()
+	_canary = new()
+	_WF = WEAKREF(_canary)
+
+/** Prime the stack overflow detector.
+	Store the return value of this proc call in a proc level var.
+	Can only be called once.
+**/
```

Comment:
The autodoc format for multiline comments is 
```
/**
 *
 *
 */
```

Reply:
the * is unneeded and annoying to maintain with alignment while writing